modules/core/src/main/scala/org/tessellation/domain/trust/storage/OrdinalTrustMap.scala

```diff
@@ -0,0 +1,34 @@
+package org.tessellation.domain.trust.storage
+
+import cats.Order
+import cats.syntax.contravariant._
+
+import org.tessellation.cli.method.ordinalTrustUpdateInterval
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.eqv
+import derevo.circe.magnolia.encoder
+import derevo.derive
+
+@derive(eqv, encoder)
+case class OrdinalTrustMap(
+  ordinal: SnapshotOrdinal,
+  peerLabels: PublicTrustMap,
+  trust: TrustMap
+)
+
+object OrdinalTrustMap {
+
+  implicit val order: Order[OrdinalTrustMap] = Order[SnapshotOrdinal].contramap(_.ordinal)
```

Comment:
is it intended to check only the ordinal when comparing two `OrdinalTrustMap`'s? right now `OrdinalTrustMap(2L, A, B) === OrdinalTrustMap(2L, C, D)` would return `true`.
That being said, always when we use the Order or Eq type class and we specify it for partial usage (like here just by the ordinal), it should be used local to the algorithm, not globally next to the class.

Reply:
Also because `Order` type class represents a total order it is used by other higher kinded types like e.g. `SortedSet` to determine which elements are equal and which are not. So if you have an implicit order like the one above available in scope where a `SortedSet` of such values is being build, then the values will get deduplicated based on that ordering. So based on the example that Marcin written above only one of these values (which one will it be is also not known, it has to be assumed nondeterministic) will be in the `SortedSet` even though we can plainly see they are not the same. Hence that's one more reason why such an order is really insecure to define as an implicit one. I would either define such an order locally Marcin suggested or at least not make it an implicit order.

Reply:
This implicit doesn't seem to be needed and has been removed.